extensions/offline-snooty/src/index.ts

```diff
@@ -1,15 +1,56 @@
 // Documentation: https://sdk.netlify.com
-import { envVarToBool, Extension } from 'util/extension';
-const extension = new Extension({
-  isEnabled: envVarToBool(process.env.OFFLINE_SNOOTY_ENABLED),
-});
-
-extension.addBuildEventHandler('onPreBuild', () => {
-  // If the build event handler is not enabled, return early
-  if (!process.env.OFFLINE_SNOOTY_ENABLED) {
-    return;
-  }
-  console.log('Hello there.');
-});
+import { NetlifyExtension } from '@netlify/sdk';
+import { convertGatsbyToHtml } from './convertGatsbyToHtml';
+import { createSnootyCopy } from './createSnootyCopy';
+import { destroyClient, uploadToS3 } from './uploadToS3';
+import {
+  type BranchEntry,
+  type DocsetsDocument,
+  type ReposBranchesDocument,
+  readEnvConfigs,
+} from './uploadToS3/utils';
+
+const extension = new NetlifyExtension();
+const NEW_SNOOTY_PATH = `${process.cwd()}/snooty-offline`;
+export const PUBLIC_OUTPUT_PATH = `${NEW_SNOOTY_PATH}/snooty/public`;
+
+// run this extension after the build and deploy are successful
+extension.addBuildEventHandler(
+  'onSuccess',
+  async ({ netlifyConfig, utils: { run } }) => {
+    // If the build event handler is not enabled, return early
+    if (!process.env.OFFLINE_SNOOTY_ENABLED) {
+      return;
+    }
+    const { bucketName, fileName } = readEnvConfigs({
+      env: netlifyConfig.build.environment.ENV ?? '',
+      docsetEntry:
+        (netlifyConfig.build.environment
+          .DOCSET_ENTRY as unknown as DocsetsDocument) ?? {},
+      repoEntry:
+        (netlifyConfig.build.environment
+          .REPO_ENTRY as unknown as ReposBranchesDocument) ?? {},
+      branchEntry:
+        (netlifyConfig.build.environment
+          .BRANCH_ENTRY as unknown as BranchEntry) ?? {},
+    });
```

Comment:
NIT: Instead of casting `as unknown as ...`, we can provide a type for the `environment` property so that we can cast as the type we want directly. 

```suggestion
    const environment = netlifyConfig.build.environment as Record<
      string,
      string | DocsetsDocument | ReposBranchesDocument | BranchEntry
    >;
    const { bucketName, fileName } = readEnvConfigs({
      env: (environment.ENV as string) ?? '',
      docsetEntry: (environment.DOCSET_ENTRY as DocsetsDocument) ?? {},
      repoEntry: (environment.REPO_ENTRY as ReposBranchesDocument) ?? {},
      branchEntry: (environment.BRANCH_ENTRY as BranchEntry) ?? {},
    });
```